assets/src/components/Sidebar/CampaignSidebar.js

```diff
@@ -86,6 +86,30 @@ export class CampaignSidebar extends Component {
       meta: null,
       parent: null,
     };
+    this.handleThemeSwitch = this.handleThemeSwitch.bind( this );
+    this.loadTheme = this.loadTheme.bind( this );
+  }
+
+  async handleThemeSwitch( metaKey, value, meta ) {
+    await this.loadTheme( value )
+    const invalidatedFields = this.state.theme.fields.filter( field => {
+      const resolvedField = getFieldFromTheme(this.state.theme, field.id, meta);
```

Comment:
We have a naming collision here, so for now called it `resolvedField`, but probably can be named better.